pytorch_lightning/plugins/environments/xla_environment.py

```diff
@@ -0,0 +1,62 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.utilities import _TPU_AVAILABLE
+
+if _TPU_AVAILABLE:
+    import torch_xla.core.xla_env_vars as xenv
+
+
+class XLAEnvironment(ClusterEnvironment):
```

Comment:
There are different ways this can be implemented:
1. Pure access to environment variable through string, i.e., `os.environ["NAME"]`
2. Use `torch_xla.core.xla_env_vars` to access the variables, i.e., ` os.environ[xenv.NAME]`
3. Use the functions in xla, i.e., `xla.xrt_world_size()`

Advantage of 1) is you can instantiate and test the plugin without requiring the xla import. 
Advantage of 2) and 3) are they are more robust to upstream changes I guess.

Reply:
I would personally go with option 3. It's just a layer on top of option 2.
And regarding option 1 reasoning, I think we could mock for better testing.